services/github/github-package-registry-downloads.service.js

```diff
@@ -0,0 +1,295 @@
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const versionDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              version: packageStatistics.allow(null),
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const latestVersionDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestVersion: packageStatistics,
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const intervalMap = {
+  today: {
+    transform: statistics => statistics.downloadsToday,
+    messageSuffix: '/today',
+  },
+  week: {
+    transform: statistics => statistics.downloadsThisWeek,
+    messageSuffix: '/week',
+  },
+  month: {
+    transform: statistics => statistics.downloadsThisMonth,
+    messageSuffix: '/month',
+  },
+  year: {
+    transform: statistics => statistics.downloadsThisYear,
+    messageSuffix: '/year',
+  },
+  total: {
+    transform: statistics => statistics.downloadsTotalCount,
+    messageSuffix: '',
+  },
+}
+
+module.exports = class GithubPackageRegistryDownloads extends GithubAuthV4Service {
+  static get category() {
+    return 'downloads'
+  }
+
+  static get route() {
+    return {
+      base: 'github/packages/downloads',
+      pattern:
+        ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version?',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub Package Registry Downloads',
+        pattern:
+          ':user/:repo/:interval(today|week|month|year|total)/:packageName',
+        namedParams: {
+          user: 'github',
+          repo: 'auto-complete-element',
+          interval: 'total',
+          packageName: 'auto-complete-element',
+        },
+        staticPreview: this.render({
+          interval: 'total',
+          count: 50,
+        }),
+        documentation,
+      },
+      {
+        title: 'GitHub Package Registry Downloads (Version)',
+        pattern:
+          ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version',
+        namedParams: {
+          user: 'github',
+          repo: 'auto-complete-element',
+          interval: 'total',
+          packageName: 'auto-complete-element',
+          version: '1.0.6',
+        },
+        staticPreview: this.render({
+          interval: 'total',
+          count: 5,
+          version: '1.0.6',
+        }),
+        documentation,
+      },
+      {
+        title: 'GitHub Package Registry Downloads (Latest Version)',
```

Comment:
This is obviously the same as the above example in that it allows the user to specify their target version, but figured I'd include this example to explicitly show users they can specify `latest`. I don't feel strongly about it though if anyone feels it should be removed